internal/source/iterator.go

```diff
@@ -246,14 +270,9 @@ func (i *Iterator) messageToRecord(msg *nats.Msg) (sdk.Record, error) {
 }
 
 // getMessagePosition returns a position of a message in the form of sdk.Position.
-func (i *Iterator) getMessagePosition(msg *nats.Msg) (sdk.Position, error) {
-	metadata, err := msg.Metadata()
-	if err != nil {
-		return nil, fmt.Errorf("get message metadata: %w", err)
-	}
-
+func (i *Iterator) getMessagePosition(msg *nats.Msg, metadata *nats.MsgMetadata) (sdk.Position, error) {
```

Comment:
```suggestion
func (i *Iterator) getMessagePosition(_ *nats.Msg, metadata *nats.MsgMetadata) (sdk.Position, error) {
```

Reply:
Or maybe just remove the param completely?